lib/bolt/core/src/context/service.rs

```diff
@@ -946,10 +941,12 @@ impl ServiceContextData {
 
 		// Expose all S3 endpoints to services that need them
 		let s3_deps = if self.depends_on_s3() {
+			// self.s3_dependencies(&run_context).await
```

Comment:
Dead code?

Reply:
Sort of, we used to have a whole dependency system thats currently short circuited to just return `true` everywhere (earlier in this file). Theres a lot of leftover code especially in this file. Might be worth cleaning up in one fell swoop